Sources/NIO/Bootstrap.swift

```diff
@@ -967,3 +999,146 @@ public final class NIOPipeBootstrap {
         }
     }
 }
+
+/// An channel option which can be applied to bootstrap using shorthand notation.
+/// - See: ServerBootstrap.serverOptions(_ options: [ShorthandBootstrapOption])
+public struct ShorthandServerBootstrapOption {
```

Comment:
I believe Johannes proposed nesting these public types within their respective bootstraps: that seems like a good idea to me.

Reply:
These structures should all be `Equatable` and `Hashable`. This will require making the enums they hold `Equatable` and `Hashable`. However, in all cases the compiler should be able to synthesise the conformance:

```swift
private enum ShorthandServerOption {
    case reuseAddr
    case autoRead
    case backlog(Int32)
}

extension ShorthandServerOption: Equatable { }

extension ShorthandServerOption: Hashable { }
```

Reply:
Done Hashable and nesting suggestions.